docs/versioning.md

```diff
@@ -0,0 +1,114 @@
+# Versioning/Release
+
+This document explains the versioning scheme and processes for this repo. The
+intent is to make it clearer how to use tags in order to select specific
+versions of the repository that are compatible with specific versions of ASM.
+
+If the only thing you need is the format and meaning of the tags, you can stop
+reading after the summary. Anything after that is a detailed description of the
+config release process.
+
+## tl;dr
+
+If you're running ASM version x.y.z-asm.w, then you should use the tag
+x.y.z-asm.w+configN with the largest N.
+
+## Summary
+
+Every commit into a release branch (ex. release-1.7-asm) will be tagged with
+the corresponding ASM version, a +, the word "config", and a positive integer N
+(ex. 1.7.3-asm3+config1) that starts with 1 and increases by one every time
+configuration is updated without an update to ASM.
+
+For example, the first commit into release-1.7-asm on version 1.7.3-asm.3 would
+be 1.7.3-asm3+config1. If another commit was made before the next ASM release,
+it would be 1.7.3-asm3+config2. After the next release, the number resets back
+to one, so if 1.7.4-asm1 was released then the first commit in the release
+branch would be 1.7.4-asm1+config1.
+
+---
+
+## Concepts
+
+* **Release branch**: a branch corresponding to a specific minor version of ASM
+that has the latest stable configuration for the latest minor release
+* **Staging branch**: a branch corresponding to a specific minor version of ASM
+that allows testing all of the configuration as one unit before merging into
+the release branch
+* **Tag**: an immutable pointer to a specific commit
+* **Major, minor, point**: their normal meaning in [SemVer](https://semver.org/)
+* **Revision**: on occasion, ASM will release an important fix without an
```

Comment:
Will this mix up the concepts between `istioctl install --revision` and the image tag, and cause confusion? The part after the hyphen are called 'pre-release' https://semver.org/#spec-item-9 in versem.

But if we are sure the revision here is referring the same idea with what we append to `istioctl install --revision`, then the doc looks good to me.

Reply:
I agree, but I don't think there's a better way to handle it. Anything after the hyphen is prerelease, but the number after "asm." specifically is a revision number. That word has been used since long before Istio was a thing: see https://docs.microsoft.com/en-us/dotnet/api/system.version?redirectedfrom=MSDN&view=netcore-3.1#remarks for an example.

I'm up for suggestions, but I think we should wait for feedback that it's an actual problem before spending time on it.